### ISSUE TITLE
decode_unicode naming

### ISSUE THREAD

**Issue by kousu** [requests/requests#4462]
In `iter_{content,lines}()`, `decode_unicode` is a misnomer. It should just be `decode`: you don't really know you're decoding from a unicode encoding. Or maybe it should be `text`, to match `response.text`: if set, it guarantees you get strings back and not bytes.

Maybe this is irrelevant if this whole feature is [getting kaiboshed](https://github.com/requests/requests/issues/3481#issuecomment-238543302), but in the meantime please consider making this more obvious.

**Comment by sigmavirus24** [comment]
> but in the meantime please consider making this more obvious.

How would we go about making it more obvious in your opinion? I'm not certain I see a clear path forward based on what has been reported.

**Comment by kousu** [comment]
My suggestion is to rename `decode_unicode` to `text`, as in "give me a text object, please", similar to how `response.text` means "give me a text object, please".

(and sorry for the lagginess, my spam filter was accidentally binning all of github, oops!)

**Comment by sigmavirus24** [comment]
Except we can't just rename function parameters.

**Comment by nateprewitt** [comment]
I agree with @sigmavirus24 here that `decode` and `text` don't really make the parameter any less ambiguous. Since we can't change this in the 2.X API and we don't seem to have clearer alternatives for the 3.0 API, I'm going to close this out. Thanks for the suggestion.
